db/bigtable.go

```diff
@@ -472,14 +529,27 @@ func (bigtable *Bigtable) SaveProposalAssignments(epoch uint64, assignments map[
 	start := time.Now()
 	ts := gcp_bigtable.Timestamp(0)
 
+	muts := make([]*gcp_bigtable.Mutation, 0, 100000)
+	keys := make([]string, 0, 100000)
+
 	for slot, validator := range assignments {
 		mut := gcp_bigtable.NewMutation()
-		mut.Set(PROPOSALS_FAMILY, fmt.Sprintf("%d", validator), ts, []byte{})
-		err := bigtable.tableBeaconchain.Apply(ctx, fmt.Sprintf("%s:e:%s:s:%s", bigtable.chainId, reversedPaddedEpoch(epoch), reversedPaddedSlot(slot)), mut)
+		mut.Set(PROPOSALS_FAMILY, "p", ts, []byte{})
```

Comment:
Because this exports only the duties for the next epoch. As we don't know at that point if a block has been proposed or not we can only mark it as not proposed (ts=0). Once it has been proposed the proper ts will be set.